File: Leetcode/3005. Count Elements With Maximum Frequency/solution.cpp
class Solution {
public:
    int maxFrequencyElements(std::vector<int>& nums) {
        std::unordered_map<int, int> freqMap;
        int maxFreq = 0;
        for (int num : nums) {
            maxFreq = std::max(maxFreq, ++freqMap[num]);
        }
        int count = 0;
        for (const auto& pair : freqMap) {
            if (pair.second == maxFreq) {
                count++;
            }
        }
        return count * maxFreq;
    }
};

File: Leetcode/3005. Count Elements With Maximum Frequency/solution.rs
impl Solution {
    pub fn max_frequency_elements(nums: Vec<i32>) -> i32 {
        let mut freq_map: HashMap<i32, i32> = HashMap::new();
        let mut max_freq = 0;
        for num in nums.iter() {
            max_freq = max_freq.max(*freq_map.entry(*num).or_insert(0) + 1);
        }
        let mut count = 0;
        for (_, &freq) in freq_map.iter() {
            if freq == max_freq {
                count += 1;
            }
        }
        count * max_freq
    }
}
